packages/bpk-component-card-list/src/BpkCardListRowRail/BpkCardListCarousel.tsx

```diff
@@ -127,30 +165,43 @@ const BpkCardListCarousel = (props: CardListCarouselProps) => {
       {children.map((card, index) => {
         if (!isValidElement(card)) return null;
 
-        const cardRefCallback = (el: HTMLDivElement | null) => {
-          cardRefs.current[index] = el;
-          observerVisibility(el, index);
-          setA11yTabIndex(el, index, visibilityList);
-        };
-
-        const cardStyle: CSSProperties = isMobile
-          ? {
-              ...shownNumberStyle,
-              visibility: renderList[index] === 1 ? 'visible' : 'hidden', // for mobile, {renderList[index] === 1 && card} will cause reflowing and bugs, implementing visibility improvement instead
-            }
-          : shownNumberStyle;
+        const cardDimensionStyle: CSSProperties = {};
+        if (firstCardWidthRef.current) {
+          cardDimensionStyle.width = `${firstCardWidthRef.current}px`;
+        }
+        if (firstCardHeightRef.current) {
+          cardDimensionStyle.height = `${firstCardHeightRef.current}px`;
+        }
+
+        // Only render cards that are within the renderList range or have been visible before
+        if (renderList[index] !== 1 && !hasBeenVisibleRef.current.has(index)) {
```

Comment:
If the current card has not entered the visible area, replace it with an empty element. The empty element should be consistent with the card width/height to avoid reflow.